closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -420,22 +470,51 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_GCMDecrypt
   jbyteArray input, jint inOffset, jint inLen, jbyteArray output, jint outOffset,
   jbyteArray aad, jint aadLen, jint tagLen) {
 
-    jboolean isCopy;
     unsigned char* inputNative;
     unsigned char* aadNative;
     int ret, len, plaintext_len = 0;
-    unsigned char* keyNative    = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, key,   &isCopy));
-    unsigned char* ivNative     = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, iv,    &isCopy));
-    unsigned char* outputNative = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, output,&isCopy));
-
+    unsigned char* keyNative;
+    unsigned char* ivNative;
+    unsigned char* outputNative;
     EVP_CIPHER_CTX* ctx = NULL;
 
+    keyNative = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, key, 0));
+    if (keyNative == NULL && keyLen > 0)
```

Comment:
Why the length checks here?

Reply:
I have checked for length to ensure that return value is not NULL for valid reasons (like if source array is NULL).

Reply:
I've removed the length checks.

Reply:
I believe if the source array is NULL, the JNI call will crash.

Reply:
Right.